engine/src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -172,6 +173,20 @@ public CompletionStage<Void> process(Message<String> message) {
                      * The Event and the Action it contains are processed by all relevant endpoint processors.
                      */
                     try {
+                        final Context context = action.getContext();
+                        if (context != null) {
+                            final var propertiesMap = context.getAdditionalProperties();
+                            if (propertiesMap != null) {
+                                final var isTestEvent = (Boolean) propertiesMap.get(TestEventConstants.TEST_ACTION_CONTEXT_TEST_EVENT);
+
+                                if (isTestEvent != null && isTestEvent) {
+                                    final UUID endpointUuid = UUID.fromString((String) propertiesMap.get(TestEventConstants.TEST_ACTION_CONTEXT_ENDPOINT_ID));
+
+                                    this.endpointProcessor.processTestEvent(event, endpointUuid);
+                                }
+                            }
+                        }
+
```

Comment:
I'm not sure we need to change anything in `EventConsumer`, see my next comment for the reason.